alchemy/src/supabase/branch.ts

```diff
@@ -0,0 +1,301 @@
+import type { Context } from "../context.ts";
+import {
+  Resource,
+  ResourceKind,
+  ResourceID,
+  ResourceFQN,
+  ResourceScope,
+  ResourceSeq,
+} from "../resource.ts";
+import { Scope } from "../scope.ts";
+import {
+  createSupabaseApi,
+  type SupabaseApiOptions,
+  type SupabaseApi,
+} from "./api.ts";
+import { handleApiError } from "./api-error.ts";
+import type { ProjectResource } from "./project.ts";
+
+/**
+ * Properties for creating or updating a Supabase Branch (preview environment)
+ */
+export interface BranchProps extends SupabaseApiOptions {
+  /**
+   * Reference to the project (string ID or Project resource)
+   */
+  project: string | ProjectResource;
+
+  /**
+   * Name of the branch
+   */
+  branchName: string;
+
+  /**
+   * Git branch to associate with this database branch (optional)
+   */
+  gitBranch?: string;
+
+  /**
+   * Whether the branch should be persistent (optional)
+   */
+  persistent?: boolean;
+
+  /**
+   * Region for the branch (optional)
+   */
+  region?: string;
+
+  /**
+   * Desired instance size (optional)
+   */
+  desiredInstanceSize?: string;
+
+  /**
+   * Release channel (optional)
+   */
+  releaseChannel?: string;
+
+  /**
+   * Whether to adopt an existing branch instead of failing on conflict
+   */
+  adopt?: boolean;
+
+  /**
+   * Whether to delete the branch on resource destruction
+   */
+  delete?: boolean;
+}
+
+/**
+ * Supabase Branch resource representing a preview environment
+ */
+export interface BranchResource extends Resource<"supabase::Branch"> {
+  /**
+   * Unique identifier of the branch
+   */
+  id: string;
+
+  /**
+   * Name of the branch
+   */
+  name: string;
+
+  /**
+   * Reference to the parent project
+   */
+  projectRef: string;
+
+  /**
+   * Reference to the parent project
+   */
+  parentProjectRef: string;
```

Comment:
projectRef should be project: string | Project

parentProjectRef should be parentProject: string | Project